src/core/db/manager.py

```diff
@@ -184,34 +404,47 @@ def add_collaborator(self, repo, collaborator, privileges):
             collaborator_obj, _ = Collaborator.objects.get_or_create(
                 user=user, repo_name=repo, repo_base=self.repo_base)
 
-        # convert privileges list to string
-        privilege_str = ', '.join(privileges)
-        collaborator_obj.permission = privilege_str
+        # convert privileges list to string and save the object
+        db_privilege_str = ', '.join(db_privileges).upper()
+        file_privilege_str = ', '.join(file_privileges).lower()
+
+        collaborator_obj.permission = db_privilege_str
+        collaborator_obj.file_permission = file_privilege_str
 
         collaborator_obj.save()
 
         return self.user_con.add_collaborator(
             repo=repo,
             collaborator=collaborator,
-            privileges=privileges
+            db_privileges=db_privileges
         )
 
     def delete_collaborator(self, repo, collaborator):
+        """
+        Removes a user's or app's privileges on a repo.
+
+        Returns True on success.
+
+        Raises LookupError when repo or collaborator does not exist.
+        Raises User.DoesNotExist if collaborator owns repo.
+        Raises PermissionDenied on insufficient permissions.
+        """
         with _superuser_connection(self.repo_base) as conn:
             collaborators = conn.list_collaborators(repo=repo)
             collaborators = [c.get('username') for c in collaborators]
 
+            # Current user must be the repo's owner or the collaborator to be
+            # removed and must be an existing collaborator.
+            if (self.username not in [collaborator, self.repo_base] or
```

Comment:
We're dealing with permissions to add collaborator and remove collaborator differently. A user should be able to remove themself as a collaborator, but in other cases, we should probably be checking to make sure the logged in user has_repo_db_privilege 'CREATE'. Or the add_collaborator should just do a simple check to make sure that `self.username==repo_base`